imports/api/employee.js

```diff
@@ -0,0 +1,234 @@
+import { Meteor } from "meteor/meteor";
+import { Employees, Tasks } from "./db";
+import EMPLOYEESAPI from "../constant/methods/employeesAPI";
+import { isAuthenticated } from "../util/authUtil";
+import { AuthError, EmployeeError } from "../constant/error";
+import { EmployeeMessage } from "../constant/message";
+import { removeElement, addToList } from "../util/arrayUtil";
+import { isAdmin, removeRelatedTasks } from "../util/employeeUtil";
+
+if (Meteor.isServer) {
+	Meteor.publish("employees", () => Employees.find());
+}
+
+export const insertEmployee = (
+	db,
+	_id,
+	employeeId,
+	firstName,
+	lastName,
+	role,
+) => {
+	if (!isAuthenticated()) {
+		throw new Meteor.Error(AuthError.NOT_AUTH);
+	}
+	if (!isAdmin(_id, Employees)) {
+		throw new Meteor.Error(EmployeeError.ADMIN_NOT_FOUND);
+	}
+
+	return db.insert(
+		{
+			employeeId,
+			firstName,
+			lastName,
+			onBoard: new Date(),
+			// keep it null for now
+			managerId: null,
+			teamId: null,
+			tasksAssienedId: [],
+			tasksCreatedId: [],
+			tasksWatchingId: [],
+			requestsSentId: [],
+			requestReceivedId: [],
+			commentsId: [],
+			filesId: [],
+			employeesId: [],
+			individualsId: [],
+			teamsId: [],
+			// role should be one of"employer", "employee", "admin"?
+			role,
+		},
+		err => {
+			if (err) {
+				throw new Meteor.Error(EmployeeError.EMPLOYEE_INSERT_FAIL);
+			} else {
+				console.log(EmployeeMessage.EMPLOYEE_CREATED);
+			}
+		},
+	);
+};
+
+export const removeEmployee = (db, _id, employeeId) => {
+	if (!isAuthenticated()) {
+		throw new Meteor.Error(AuthError.NOT_AUTH);
+	}
+	if (!isAdmin(_id, Employees)) {
+		throw new Meteor.Error(EmployeeError.ADMIN_NOT_FOUND);
+	}
+	const employee = Employees.findOne({ employeeId });
+	const removedId = employee.employeeId;
+	if (!employee) {
+		throw new Meteor.Error(EmployeeError.EMPLOYEE_NOT_EXIST);
+	}
+	if (removedId !== this.employeeId) {
+		throw new Meteor.Error(AuthError.NO_PRIVILEGE);
+	}
+
+	return db.remove(
+		{
+			employeeId,
+		},
+		err => {
+			if (err) {
+				throw new Meteor.Error(EmployeeError.EMPLOYEE_REMOVE_FAIL);
+			} else {
+				let { tasksAssienedId } = employee;
+				tasksAssienedId = [...tasksAssienedId];
+				removeRelatedTasks(tasksAssienedId, employeeId, Tasks);
+				
+				let { tasksCreatedId } = employee;
+				tasksCreatedId = [...tasksCreatedId];
+				removeRelatedTasks(tasksCreatedId, employeeId, Tasks);
+
+				let { tasksWatchingId } = employee;
+				tasksWatchingId = [...tasksWatchingId];
+				removeRelatedTasks(tasksWatchingId, employeeId, Tasks);
+			}
+		},
+	);
+};
+
+export const createTask = (db, _id, taskId) => {
+	if (!this.userId) {
+		throw new Meteor.Error(AuthError.NOT_AUTH);
+	}
+	// TODO: not too sure use _id or employeeid
+	const employee = db.findOne({ _id });
+	if (!employee) {
+		throw new Meteor.Error(EmployeeError.EMPLOYEE_NOT_EXIST);
+	}
+	let { tasksCreatedId } = employee;
+	tasksCreatedId = [...tasksCreatedId];
+	if (!tasksCreatedId.includes(taskId)) {
```

Comment:
Let's add the existence check to the addToList helper, seems like something we will be checking a lot.

Reply:
we only return `db.update` when `taskid` is not included in the current `taskscreatedid`, therefore i dont think it save time to add this to the helper cuz we have to check it for `db.update` either way.

Reply:
Makes sense, let's leave it then.